views/pages/tasks.ejs

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <%- include('../partials/head'); %>
+</head>
+<body class="container">
+
+<header>
+  <%- include('../partials/header'); %>
+</header>
+
+<main>
+  <p><%= message %></p>
+  <div class="jumbotron">
+    <% if (tasks.length > 0) { %>
+      <table class="table"
+        <tr><th>Name</th><th>Completed</th><th colspan="2"></th></tr>
+        <% tasks.forEach(function(task){ %>
+          <tr><td><%= task.name %></td><td><%= task.completed %></td>
```

Comment:
Extra challenge: can we add a "mark as complete" or "mark as incomplete" button for each task depending on whether its currently complete, which will toggle its completion state? As always, please ask if you have questions about how to do this!

Reply:
Do you think we really gonna use ejs anywhere? I use react.js or next.js as my front end normally.

Reply:
Haha it's true that react is definitely the most popular for any actively developed commercial web project today by a wide margin. However, server-side templating libraries like this are pretty common, and if you look at any old ruby on rails, Django, or even PHP project, it uses this type of architecture with server-rendered templates and shipping plain html to the client. Actually, the pendulum is swinging back towards shipping plain HTML to the client as a trendy practice. This is best exemplified by the new remix.run react meta-framework.

Reply:
I got this code but its not working however the Frontend looks good:

`<td><button type="submit" name="completed" value ="{{!task.completed}}" class="btn btn-primary" onclick=<%= "/tasks/update/" + task.id %>  >
              <% if (task.completed) { %>
                Mark as incompleted
                <% } else { %>
                Mark as completed
                <% } %>
                </button></td>`

Reply:
So, the trick for a "button" that flips the `completed` state is for it to actually be a form like this:

```
<form action=<%= "/tasks/update/" + task.id %> method="post">
  <input type="hidden" name="completed" value=<%= !task.completed %>
  <button class="btn bin-secondary">Mark <%= task.completed? 'complete' : 'incomplete' %>
</form>
```

That trick of putting a `hidden` input with a hard-coded value is handy because we can create "pre-filled forms" that users can trigger with the click of a button.

Reply:
I get that, but why would the value of the button wont pass if I have it like this.

`<button type="submit" name="completed" value = <%= !task.completed %> class="btn btn-primary">`

with the above ternary operator The button is not accessible.

![image](https://user-images.githubusercontent.com/81403558/188494935-bc1b6bc7-597c-4e73-89d9-22dc0c14d12a.png)

Reply:
The code is working but am not able to get the button class working to show the button same way as the Edit and Delete button

Reply:
Ok nice, that looks good! I'm not sure if you used the button with a value or the hidden field. If you used the button with the value, that's cool -- I didn't know that would work!

In general, you might prefer to use hidden fields if you want a button to trigger sending more than one piece of data since you can have as many form fields as you want in that case.

I'm glad to see you got this working though, awesome work!